public/app/plugins/panel/canvas/editor/element/utils.ts

```diff
@@ -85,8 +86,14 @@ const getData = (api: APIEditorConfig) => {
   return data;
 };
 
-const requestMatchesGrafanaOrigin = (requestEndpoint: string) => {
+const isRequestUrlAllowed = (requestEndpoint: string) => {
+  if (config.actions.allowPostURL !== '') {
```

Comment:
is api.endpoint only used for POST/PUT commands?

Reply:
no, it's for everything. I removed the method check as I wanted to re-confirm that we're going to allow GET in any situation. If we're confident then I'll add it back.

This PR still needs a check against what we plan to support/restrict. But I also think these validations make more sense to happen in the backend.

Reply:
@ryantxu what's the value of exposing the url pattern in the frontend? is it just so we can validate it before we send the actual request and error early?

Reply:
if every request is proxied though the backend, it could be on the backend

Reply:
> But I also think these validations make more sense to happen in the backend.

or are you asking if we can expose a new backend service that validates the request and then make the request from the frontend?

Reply:
> or are you asking if we can expose a new backend service that validates the request and then make the request from the frontend?

no def not!

> if every request is proxied though the backend, it could be on the backend

if the main concern is actually protecting grafana endpoints, then it has to be done on the backend to work "for real". for protecting external endpoints, well, we cannot _actually_ protect them. so this JS-side guard would be more of a footgun safety. that still has value, but also requires that the backend exposes its possibly-misconfigured weak regexps, and more time spent on this PR.

given that we have a tight deadline to get this merged, i'd rather we focus on MVP that guards the Grafana instance (backend auth middleware) without also figuring out the footgun helper right now.